src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java

```diff
@@ -168,36 +171,43 @@ public Claim saveClaim(
     ) {
         String externalId = claimData.getExternalId().toString();
-
-        Long prePaymentClaimId = caseRepository.getOnHoldIdByExternalId(externalId, authorisation);
-
-        LocalDateTime now = LocalDateTimeFactory.nowInLocalZone();
         Optional<GeneratePinResponse> pinResponse = Optional.empty();
+        UserDetails userDetails = userService.getUserDetails(authorisation);
 
         if (!claimData.isClaimantRepresented()) {
             pinResponse = Optional.of(userService.generatePin(claimData.getDefendant().getName(), authorisation));
         }
 
-        Optional<String> letterHolderId = pinResponse.map(GeneratePinResponse::getUserId);
-        LocalDate issuedOn = issueDateCalculator.calculateIssueDay(now);
-        LocalDate responseDeadline = responseDeadlineCalculator.calculateResponseDeadline(issuedOn);
-        UserDetails userDetails = userService.getUserDetails(authorisation);
-        String submitterEmail = userDetails.getEmail();
-
-        Claim claim = Claim.builder()
-            .id(prePaymentClaimId)
-            .claimData(claimData)
-            .submitterId(submitterId)
-            .issuedOn(issuedOn)
-            .responseDeadline(responseDeadline)
-            .externalId(externalId)
-            .submitterEmail(submitterEmail)
-            .createdAt(nowInUTC())
-            .letterHolderId(letterHolderId.orElse(null))
-            .features(features)
-            .build();
+        Claim issuedClaim;
+        try {
+            Long prePaymentClaimId = caseRepository.getOnHoldIdByExternalId(externalId, authorisation);
+
+            LocalDateTime now = LocalDateTimeFactory.nowInLocalZone();
+
+            Optional<String> letterHolderId = pinResponse.map(GeneratePinResponse::getUserId);
+            LocalDate issuedOn = issueDateCalculator.calculateIssueDay(now);
+            LocalDate responseDeadline = responseDeadlineCalculator.calculateResponseDeadline(issuedOn);
+            String submitterEmail = userDetails.getEmail();
+
+            Claim claim = Claim.builder()
+                .id(prePaymentClaimId)
+                .claimData(claimData)
+                .submitterId(submitterId)
+                .issuedOn(issuedOn)
+                .responseDeadline(responseDeadline)
+                .externalId(externalId)
+                .submitterEmail(submitterEmail)
+                .createdAt(nowInUTC())
+                .letterHolderId(letterHolderId.orElse(null))
+                .features(features)
+                .build();
 
-        Claim issuedClaim = caseRepository.saveClaim(authorisation, claim);
+            issuedClaim = caseRepository.saveClaim(authorisation, claim);
+        } catch (ConflictException e) {
+            appInsights.trackEvent(AppInsightsEvent.CLAIM_ATTEMPT_DUPLICATE, CLAIM_EXTERNAL_ID, externalId);
+            issuedClaim = caseRepository.getClaimByExternalId(externalId, authorisation)
+                .orElseThrow(RuntimeException::new);
```

Comment:
Prefer not to throw a generic exception, how about `NotFoundException` instead of `RuntimeException`?

Reply:
Yes, good spot, but not sure which exception is most appropriate. In terms of business logic this shouldn't be possible - only if the `caseRepository` somehow lost the claim between calling `getOnHoldIdByExternalId` and here. I'll go with `NotFoundException` for now.

Reply:
Wouldn't a `NotFoundException` bubble back as a 404? I think a 500 would be more appropriate for the user

Reply:
We don't display these HTTP errors to users, we have generic error page `error.njk` in prod. We will see this in our logs only.